### ISSUE TITLE
[osx] homebrew: Error: Cannot link cmake 

### ISSUE THREAD

**Issue by pat-s** [travis-ci/travis-ci#7664]
Trying to install `qgis2-ltr` from the `osgeo` tap and getting this `cmake` link error during installation:

```
==> Installing osgeo/osgeo4mac/qgis2-ltr dependency: python
==> Downloading https://homebrew.bintray.com/bottles/python-2.7.13.sierra.bottle
==> Pouring python-2.7.13.sierra.bottle.1.tar.gz
==> /usr/local/Cellar/python/2.7.13/bin/python -s setup.py --no-user-cfg install
==> /usr/local/Cellar/python/2.7.13/bin/python -s setup.py --no-user-cfg install
==> /usr/local/Cellar/python/2.7.13/bin/python -s setup.py --no-user-cfg install
==> Caveats
Pip and setuptools have been installed. To update them
  pip install --upgrade pip setuptools
You can install Python packages with
  pip install <package>
They will install into the site-package directory
  /usr/local/lib/python2.7/site-packages
See: http://docs.brew.sh/Homebrew-and-Python.html
==> Summary
🍺  /usr/local/Cellar/python/2.7.13: 3,526 files, 48MB
Error: Cannot link cmake
Another version is already linked: /usr/local/Cellar/cmake/3.7.1
```

What I tried so far:
* different `xcode` images 
* unlinking `cmake` | reinstall `cmake` (both permission denied)

**Comment by BanzaiMan** [comment]
How does this behave on your machine? You are using some nonstandard tap, and it is doing something to fiddle with the existing installation of `cmake`. You should study the formula to see what it is trying to do.

**Comment by pat-s** [comment]
`brew edit qgis2-ltr`:

54:  `depends_on "cmake" => :build`

that's what I see after a first look - does this line have potential to cause the problem? 

ping @dakcarto

**Comment by pat-s** [comment]
using `xcode8.3` image I get the some lines further and then face the following error:

```ruby
Pouring unixodbc-2.3.4.sierra.bottle.1.tar.gz
Error: The `brew link` step did not complete successfully
The formula built, but is not symlinked into /usr/local
Could not symlink share/man/man7/unixODBC.7
/usr/local/share/man/man7 is not writable.
You can try again using:
  brew link unixodbc
```

**Comment by dakcarto** [comment]
@pat-s can you share your code repo where the `.travis.yml` file is? You are installing `qgis2-ltr` so it must be something custom you have set up, yes?

**Comment by pat-s** [comment]
@dakcarto sure, here is the yaml: https://github.com/jannes-m/RQGIS/blob/reticulate-travis/.travis.yml

**Comment by dakcarto** [comment]
To work around the permission errors, try just setting ownership of /usr/local to your user, e.g. `sudo chown -R $(whoami) /usr/local` before any `brew` commands.

For the line:
```
if [[ "${LTR_mac}" ]]; then brew install qgis2-ltr --with-saga-gis -with-grass; fi
```
try adding verbosity, e.g. `... brew install -v qgis2-ltr ...` to see if it sheds more light on the issue (it may not).

Odd thing, when using `--with-saga-gis -with-grass`, those should show up as dependencies needing installed prior to `qgis2-ltr`, not just `python`. Do they, but you just clipped the output at `python`?

**Comment by pat-s** [comment]
> Odd thing, when using --with-saga-gis -with-grass, those should show up as dependencies needing installed prior to qgis2-ltr, not just python. Do they, but you just clipped the output at python?

I clipped all other dependency installations prior to `python`, I think grass and saga showed up yes (but no confirmation). 

I am trying to get ownership of `/usr/local/` right now and see if that helps. Thanks for your quick response!

**Comment by pat-s** [comment]
> To work around the permission errors, try just setting ownership of /usr/local to your user, e.g. sudo chown -R $(whoami) /usr/local before any brew commands.

Helped and solved the installation problem of dependencies. 

Now back at a `travis-ci` problem (I assume):

``` ruby
if [[ "${LTR_mac}" ]]; then travis_wait 100 brew install -v qgis2-ltr --with-saga-gis -with-grass; fi
if [[ "${DEV_mac}" ]]; then travis_wait 100 brew install -v qgis2 --with-saga-gis -with-grass; fi

Still running (46 of 100): brew install -v qgis2 --with-saga-gis -with-grass
The job exceeded the maximum time limit for jobs, and has been terminated.
```

Build was terminated after 47 min although I have `travis_wait 100`? On my local machine building takes around 30 - 40 min - so I'm wondering what causes the long build times on travis?

Log: https://travis-ci.org/jannes-m/RQGIS/jobs/226320624

**Comment by pat-s** [comment]
Okay it seems this also happened to normal QGIS builds on travis: http://osgeo-org.1560.x6.nabble.com/Travis-The-job-exceeded-the-maximum-time-limit-for-jobs-and-has-been-terminated-td5301763.html

I think there is no way that my build job time is also increased?

**Comment by BanzaiMan** [comment]
We have a hard time limit of 50 minutes for public builds on the Mac at the moment.

**Comment by pat-s** [comment]
Ok, thanks for the information. I close this issue for now since the current limitation seems to be the public build time limit and not anymore the initial `cmake` problem (which was solved by using `xcode8.3`)

**Comment by dakcarto** [comment]
@pat-s adding options to the install of qgis2 formulae will cause them to be compiled from source. Since the SAGA and GRASS options are for convenience only, you can install those first, then QGIS.In this way you get the benefit on utilizing bottled binaries for all of them, instead of compiling.

**Comment by pat-s** [comment]
@dakcarto ahh, I did not know that! That speeded up the installation and helped me not hitting the time-out, thanks a lot! I'm now struggling with some other `gdal` probs but they are unrelated to this issue here.

**Comment by MaffooBristol** [comment]
My two pence:

This is the only result on google for "cannot link cmake", so I'll add my use-case here for those who it may help.

My issue wasn't with travis, but with installing PHP 7.0 on Mavericks. In doing so, I broke bash, which required me to `brew upgrade bash`. This, then, broke `cmake` itself. I actually had to do `sudo chown -R matt /usr/local/share/cmake` and then `brew upgrade cmake`, which made future brew installations work. So it can be a bit of a walk around the park to get it all working, but I don't think it's an issue related to the package being installed, just out of date brew dependencies!
